### ISSUE TITLE
Mounting DMG fails because of sandboxing

### ISSUE THREAD

**Issue by Iomegan** [sparkle-project/Sparkle#1147]
Hi,

I am trying to integrate the [sandbox comparable version of Sparkle](https://github.com/sparkle-project/Sparkle/tree/ui-separation-and-xpc) into my app. Everything works fine so far. The update even works if the app is not sandboxed. If it is sandboxed, it stops at the point where it tries to mount the DMG that contains the updated app.

I see the following messages in the console:
```
sandboxd	Sandbox: hdiutil(15445) deny sysctl-write vfs.generic.noremotehang
Violation:       deny sysctl-write vfs.generic.noremotehang 
MetaData: {"build":"Mac OS X 10.13.1 (17B25c)","action":"deny","target":["vfs.generic.noremotehang"],"hardware":"Mac","platform_binary":"no","profile":"unknown","process":"hdiutil","op":"sysctl-write"}

Process:         hdiutil [15445]
Path:            /usr/bin/hdiutil
Load Address:    0x101325000
Identifier:      hdiutil
Version:         ??? (???)
Code Type:       x86_64 (Native)
Parent Process:  My App[15403]
Responsible:     /Users/username/Desktop/Release Paddle 2017-10-30 4-59-19 PM/My App.app/Contents/MacOS/My App [15403]
User ID:         501

Date/Time:       2017-10-30 17:36:59.438 GMT+10
OS Version:      Mac OS X 10.13.0
Report Version:  8

Thread 0 (id: 458459):
0   libsystem_kernel.dylib        	0x00007fff5a798532 __sysctlbyname + 10
1   DiskImages                    	0x00007fff4741d768 DIInitialize + 147
2   hdiutil                       	0x00000001013267e8
3   libdyld.dylib                 	0x00007fff5a648144 start + 0
4   hdiutil                       	0x0000000000000007

Binary Images:
       0x101325000 -        0x101366ffb  hdiutil (480.1.2) <c9527b5a-f0ac-378f-825b-f0d881ca43af> /usr/bin/hdiutil
    0x7fff4741a000 -     0x7fff474ecff7  com.apple.DiskImagesFramework (480.1.2 - 480.1.2) <67132748-9a93-384f-a656-024c5ac860ee> /System/Library/PrivateFrameworks/DiskImages.framework/Versions/A/DiskImages
    0x7fff5a647000 -     0x7fff5a664ff7  libdyld.dylib (519.2.1) <da122f3c-723f-3088-9c30-ffd63464fd52> /usr/lib/system/libdyld.dylib
    0x7fff5a77c000 -     0x7fff5a7a1ff7  libsystem_kernel.dylib (4570.20.55) <e4f4b26d-0535-38b1-963e-f47bfec586cf> /usr/lib/system/libsystem_kernel.dylib
```
and 
```My App	hdiutil failed with code: 1 data: <<>>```
and
```My App	Can't mount DMG /Users/username/Library/Containers/com.mac-attender.PDF-Squeezer/Data/Library/Caches/com.example.My-App/org.sparkle-project.Sparkle/My App 116b/My App.dmg```

The dmg file has been successfully signed with my apple ID.  (tested with `codesign --verify`).

Not sure what the issue is here. Maybe I've missed a step. I tried to find the mysterious INSTALL file that is mentioned in the Read Me so I could check all steps that are needed for the Sandbox comparable version of Sparkle - but looks like it does not exist yet.

The issue is reproducible on all machines I've tested from 10.9 to 10.13.

**Comment by pornel** [comment]
Does this work with zip/tarball? 

Our implementation of DMG support is an ugly hack, so I'd recommend to avoid that format if at all possible.

**Comment by Iomegan** [comment]
Unfortunately ZIP also fails:

`My App	Failed to release quarantine on /Users/username/Library/Containers/com.example.My-App/Data/Library/Caches/com.exampleMy -pp/org.sparkle-project.Sparkle/My App (Autoupdate).app with error Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" UserInfo={NSLocalizedDescription=Failed to remove file quarantine on My App (Autoupdate).app.}`

My App has been signed with my developer ID and RB App Checker reports that everything is okay with it. The zip file i also signed: `codesign -s "Developer ID Application: My Name" --keychain "/Users/username/Library/Keychains/login.keychain" "MyApp.zip"` and successfully verified. I can also download the zip and open the app without translocation / Gatekeeper complaining. Hmm...

**Comment by Iomegan** [comment]
I've found those XPC services in other app's integration of Sparkle:
`org.sparkle-project.InstallerConnection.xpc
org.sparkle-project.InstallerLauncher.xpc
org.sparkle-project.InstallerStatus.xpc`

However I don't find them in the current repository. Is that correct? They are also not being build with _make release_.

**Comment by pornel** [comment]
It may be incorrect. `make release` was written for sparkle 1.x

**Comment by Iomegan** [comment]
Thanks pornel, I have found the issue. I was not using the [Sparkle-XPC-UI branch](https://github.com/sparkle-project/Sparkle/tree/ui-separation-and-xpc), I was using the master branch. The reason for that is that I clicked _Open in Xcode_ in the Sparkle-XPC-UI branch, which clones the master. So manually downloading it got me the correct branch with all the XPC services. `make release` also works as expected with that branch.

Thanks for your help!
